packages/web-client/app/models/workflow/workflow-session.ts

```diff
@@ -1,14 +1,37 @@
+import WorkflowPersistence from '@cardstack/web-client/services/workflow-persistence';
 import { tracked } from '@glimmer/tracking';
 
 export interface ArbitraryDictionary {
   [key: string]: any;
 }
+
 export default class WorkflowSession {
+  workflow: any;
+  workflowPersistenceStorage: WorkflowPersistence | undefined;
+
+  constructor(workflow?: any) {
+    this.workflow = workflow;
+
+    // Allow WorkflowSession to be instantiated as a POJO (for unit tests)
+    if (this.workflow && typeof this.workflow.owner.lookup === 'function') {
```

Comment:
Is there a reason for this guard? Wouldn't the owner always have a lookup method?

Reply:
Added this condition because of `new ConcreteWorkflow({});` in tests (where lookup isn't a function on the `{}`). I think it's better to remove this guard and change all `new ConcreteWorkflow({})` to `new ConcreteWorkflow(this.owner)`

Reply:
OK, this seems fine. There is a way we could refactor to make our container construct our workflow instances and allow injections to work. I can pair with you on it for a future PR.